File: api/src/endpoint/question/create_answer.rs
use serde::{Deserialize, Serialize};
use api_derive::EndpointRequest;

use crate::{endpoint::Endpoint, resource::question::AnswerResource};

pub struct CreateAnswer;

impl Endpoint for CreateAnswer {
    const PATH: &'static str = "/repository/{owner}/{repo}/question/{number}/answer";
    const METHOD: http::Method = http::Method::POST;

    type Request = CreateAnswerRequest;
    type Response = CreateAnswerResponse;
}

#[derive(EndpointRequest, Debug, Serialize, Deserialize)]
pub struct CreateAnswerRequest {
    pub body: String,
}

pub type CreateAnswerResponse = AnswerResource;

File: gitdot-api-ts/src/endpoint/question/create-answer.ts
import { z } from "zod";

import { AnswerResource } from "../../resource";

export const CreateAnswerRequest = z.object({
  body: z.string(),
});
export type CreateAnswerRequest = z.infer<typeof CreateAnswerRequest>;

export const CreateAnswer = {
  path: "/repository/{owner}/{repo}/question/{number}/answer",
  method: "POST",
  request: CreateAnswerRequest,
  response: AnswerResource,
} as const;
export type CreateAnswer = typeof CreateAnswer;
